packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts

```diff
@@ -166,12 +170,67 @@
     this._session = session
   }
 
-  async subscribeToAppLogs(input: AppLogsSubscribeVariables): Promise<AppLogsSubscribeResponse> {
-    throw new Error(`Not Implemented: ${JSON.stringify(input)}`)
+  async subscribeToAppLogs(
+    input: AppLogsSubscribeMutationVariables,
+    organizationId: string,
+  ): Promise<AppLogsSubscribeMutation> {
+    const token = await this.token()
+
+    return appManagementRequestDoc<AppLogsSubscribeMutation, AppLogsSubscribeMutationVariables>(
+      organizationId,
+      AppLogsSubscribe,
+      token,
+      {
+        shopIds: input.shopIds,
+        apiKey: input.apiKey,
+      },
+    )
   }
 
-  async appLogs(options: AppLogsOptions): Promise<AppLogsResponse> {
-    throw new Error(`Not Implemented: ${JSON.stringify(options)}`)
+  async appLogs(
+    options: {
+      jwtToken: string
+      cursor?: string
+      filters?: {
+        status?: string
+        source?: string
+      }
+    },
+    organizationId: string,
+  ): Promise<AppLogsResponse> {
+    const response = await fetchAppLogs({
+      organizationId,
+      jwtToken: options.jwtToken,
+      cursor: options.cursor,
+      filters: options.filters,
+    })
+
+    try {
+      const data = (await response.json()) as {
+        app_logs?: AppLogData[]
+        cursor?: string
+        errors?: string[]
+      }
+
+      if (!response.ok) {
+        return {
+          errors: data.errors || [`Request failed with status ${response.status}`],
```

Comment:
Can a response be not-ok and still have data?

Reply:
Yes, our [backend controller](https://github.com/Shopify/partners/blob/main/components/app_logs/app/controllers/app_logs/app_logs_controller.rb#L104-L129) returns error data with non-200 status codes in a bunch of cases. 

Then the error messages are used by the client to determine retry, display appropriate messages, and adjust polling intervals.